test/unit/specs/components/governance/ModalVote.spec.js

```diff
@@ -20,74 +20,73 @@ describe(`ModalVote`, () => {
         proposalTitle: lcdClientMock.state.proposals[`1`].title
       }
     })
+
+    instance.store.state.connection.connected = true
     wrapper = instance.wrapper
-  })
 
-  describe(`component matches snapshot`, () => {
-    it(`has the expected html structure`, async () => {
-      expect(wrapper.vm.$el).toMatchSnapshot()
-    })
+    wrapper.vm.$refs.actionModal.submit = jest.fn(cb => cb())
   })
 
-  describe(`default values are set correctly`, () => {
-    it(`the 'option' defaults to an empty string`, () => {
-      expect(wrapper.vm.option).toEqual(``)
-    })
-
-    it(`account password defaults to an empty string`, () => {
-      expect(wrapper.vm.password).toEqual(``)
-    })
-
-    it(`password is hidden by default`, () => {
-      expect(wrapper.vm.showPassword).toBe(false)
-    })
+  it(`has the expected html structure`, async () => {
+    expect(wrapper.vm.$el).toMatchSnapshot()
   })
 
-  describe(`Password display`, () => {
-    it(`toggles the password between text and password`, () => {
-      wrapper.vm.togglePassword()
-      expect(wrapper.vm.showPassword).toBe(true)
-      wrapper.vm.togglePassword()
-      expect(wrapper.vm.showPassword).toBe(false)
-    })
-  })
+  describe(`submits form only if inputs are correct`, () => {
+    it(`does not submit in cases`, async () => {
+      wrapper.vm.submitForm = jest.fn()
 
-  describe(`enables or disables Vote correctly`, () => {
-    it(`disables the 'Vote' button`, () => {
       // default values
-      let voteBtn = wrapper.find(`#cast-vote`)
-      expect(voteBtn.html()).toContain(`disabled="disabled"`)
+      await wrapper.vm.$nextTick()
+      wrapper.vm.validateForm()
```

Comment:
every instance of 
`wrapper.vm.validateForm()` 
should become 
`await wrapper.vm.validateForm()`

Reply:
opting for not doing this as the tested functionality is synchronous and in the following PR `validateForm` will be synchronous